tests/functional/test_credentials.py

```diff
@@ -472,6 +478,37 @@ def test_process_source_profile(self):
         }
         self.assertEqual(kwargs, expected_kwargs)
 
+    def test_web_identity_source_profile(self):
+        token_path = os.path.join(self.tempdir, 'token')
+        with open(token_path, 'w') as token_file:
+            token_file.write('a.token')
+        config = (
+            '[profile A]\n'
+            'role_arn = arn:aws:iam::123456789:role/RoleA\n'
+            'source_profile = B\n'
+            '[profile B]\n'
+            'role_arn = arn:aws:iam::123456789:role/RoleB\n'
+            'web_identity_token_file = %s\n' % token_path
+        )
+        self.write_config(config)
+
+        session, stubber = self.create_session(profile='A')
+
+        identity_creds = self.create_random_credentials()
+        identity_response = self.create_assume_role_response(identity_creds)
+        stubber.add_response(
+            'assume_role_with_web_identity',
+            identity_response,
+        )
+
+        expected_creds = self.create_random_credentials()
+        assume_role_response = self.create_assume_role_response(expected_creds)
+        stubber.add_response('assume_role', assume_role_response)
+
+        actual_creds = session.get_credentials()
+        self.assert_creds_equal(actual_creds, expected_creds)
```

Comment:
We should make sure to assert that we are creating the sts client with the correct credentials similar to how we do it for the process provider.